op-node/rollup/derive/l1_block_info.go

```diff
@@ -160,10 +165,32 @@ func (info *L1BlockInfo) unmarshalBinaryBedrock(data []byte) error {
 // | 32      | BatcherHash              |
 // +---------+--------------------------+
 
+func (info *L1BlockInfo) marshalBinaryIsthmus() ([]byte, error) {
+	out, err := marshalBinaryEcotoneOrIsthmus(info, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal Isthmus l1 block info: %w", err)
+	}
+	return out, nil
+}
+
 func (info *L1BlockInfo) marshalBinaryEcotone() ([]byte, error) {
-	w := bytes.NewBuffer(make([]byte, 0, L1InfoEcotoneLen))
-	if err := solabi.WriteSignature(w, L1InfoFuncEcotoneBytes4); err != nil {
-		return nil, err
+	out, err := marshalBinaryEcotoneOrIsthmus(info, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal Ecotone l1 block info: %w", err)
+	}
+	return out, nil
+}
+
+func marshalBinaryEcotoneOrIsthmus(info *L1BlockInfo, isIsthmus ...bool) ([]byte, error) {
+	w := bytes.NewBuffer(make([]byte, 0, L1InfoEcotoneLen)) // Ecotone and Isthmus have the same length
+	if isIsthmus != nil && isIsthmus[0] {
```

Comment:
Is the variadic function argument meant to create an optional parameter? It would be nice to just use a regular bool arg, and simply set it to false pre-interop.

Reply:
agreed, this was a carry-on from a previous over-loading function.